core/src/test/java/org/elasticsearch/cluster/metadata/IndexCreationTaskTests.java

```diff
@@ -0,0 +1,450 @@
+            .putAlias(AliasMetaData.builder("alias1").searchRouting("2").build())
+        );
+        addMatchingTemplate(builder -> builder
+            .order(1)
+            .settings(Settings.builder().put(SETTING_NUMBER_OF_SHARDS, 10))
+            .putAlias(AliasMetaData.builder("alias1").searchRouting("1").build())
+        );
+        final ClusterState result = executeTask();
+
+        assertEquals("12", result.getMetaData().index("test").getSettings().get(SETTING_NUMBER_OF_SHARDS));
+        assertEquals("3", result.metaData().index("test").getAliases().get("alias1").getSearchRouting());
+    }
+
+    public void testRequestStateOpen() throws Exception {
+
+        when(request.state()).thenReturn(IndexMetaData.State.OPEN);
+
+        executeTask();
+
+        verify(allocationService, times(1)).reroute(anyObject(), anyObject());
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testIndexRemovalOnFailure() throws Exception {
+        doThrow(new RuntimeException("oops")).when(mapper).merge(anyMap(), anyObject(), anyBoolean());
+
+        try {
+            executeTask();
+            fail("exception not thrown");
+        } catch (RuntimeException e) {
+            verify(indicesService, times(1)).removeIndex(anyObject(), anyObject(), anyObject());
+        }
+    }
+
+    public void testShrinkIndexIgnoresTemplates() throws Exception {
+        final Index source = new Index("source_idx", "aaa111bbb222");
+
+        when(request.shrinkFrom()).thenReturn(source);
+
+        currentStateMetaDataBuilder.put(createIndexMetaDataBuilder("source_idx", "aaa111bbb222", 2, 2));
+
+        routingTableBuilder.add(createIndexRoutingTableWithStartedShards(source));
+
+        when(currentStateBlocks.indexBlocked(eq(ClusterBlockLevel.WRITE), eq("source_idx"))).thenReturn(true);
+        reqSettings.put(SETTING_NUMBER_OF_SHARDS, 1);
+
+        addMatchingTemplate(builder -> builder
+            .putAlias(AliasMetaData.builder("alias1").searchRouting("fromTpl").build())
+            .putMapping("mapping1", createMapping())
+            .putCustom("custom1", createCustom())
+            .settings(Settings.builder().put("key1", "tplValue"))
+        );
+
+        final ClusterState result = executeTask();
+
+        assertFalse(result.metaData().index("test").getAliases().containsKey("alias1"));
+        assertFalse(result.metaData().index("test").getCustoms().containsKey("custom1"));
+        assertNull(result.metaData().index("test").getSettings().get("key1"));
+        assertFalse(getMappingsFromResponse().containsKey("mapping1"));
+    }
+
+    public void testValidateWaitForActiveShardsFailure() throws Exception {
+        waitForActiveShardsNum = ActiveShardCount.from(1000);
+
+        try {
+            executeTask();
+            fail("validation exception expected");
+        } catch (IllegalArgumentException e) {
+            assertTrue(e.getMessage().contains("invalid wait_for_active_shards"));
+        }
```

Comment:
Can you rewrite this block using `expectThrows`? Also, instead of `assertTrue`, it's more effective to use the built-in matchers like `assertThat(e.getMessage(), hasToString(containsString("invalid wait_for_active_shards"));`. The reason for this is because if the assertion fails with `assertTrue`, the failure message only says something like "expected true but was false" where as with the matchers we get something like "expected \"invalid wait_for_active_shards\" but was ..." so we already have immediately from the failure message more information about what is happening.